src/controllers/controller.line.js

```diff
@@ -181,29 +179,29 @@ module.exports = DatasetController.extend({
 		var me = this;
 		var chart = me.chart;
 		var yScale = me._yScale;
+		var xScale = me._xScale;
+		var xValue = me._getParsedValue(index, xScale);
 		var sumPos = 0;
 		var sumNeg = 0;
-		var i, ds, dsMeta;
+		var i, dsMeta, stackedValue;
 
 		if (yScale.options.stacked) {
 			for (i = 0; i < datasetIndex; i++) {
-				ds = chart.data.datasets[i];
 				dsMeta = chart.getDatasetMeta(i);
 				if (dsMeta.type === 'line' && dsMeta.yAxisID === yScale.id && chart.isDatasetVisible(i)) {
-					var stackedRightValue = Number(yScale.getRightValue(ds.data[index]));
-					if (stackedRightValue < 0) {
-						sumNeg += stackedRightValue || 0;
+					stackedValue = dsMeta.controller._getParsedValueByScaleValue(xValue, xScale);
```

Comment:
Why do we need to use the `xScale` and `xValue` here?  Before it used the `yScale` and `index` which makes more sense to me